diff --git a/src/objects/objects.cc b/src/objects/objects.cc
index 1156a433e3b..2591fb6ec86 100644
--- a/src/objects/objects.cc
+++ b/src/objects/objects.cc
@@ -2316,12 +2316,6 @@ Maybe<bool> Object::SetPropertyInternal(LookupIterator* it,
               // Proceed lookup.
               break;
           }
-          // Assuming that the callback has side effects, we use
-          // Object::SetSuperProperty() which works properly regardless on
-          // whether the property was present on the receiver or not when
-          // storing to the receiver.
-          // Proceed lookup from the next state.
-          it->Next();
         } else {
           Maybe<PropertyAttributes> maybe_attributes =
               JSObject::GetPropertyAttributesWithInterceptor(it);
@@ -2342,8 +2336,10 @@ Maybe<bool> Object::SetPropertyInternal(LookupIterator* it,
             // property to the receiver.
             it->NotFound();
           }
+          return Object::SetSuperProperty(it, value, store_origin,
+                                          should_throw);
         }
-        return Object::SetSuperProperty(it, value, store_origin, should_throw);
+        break;
       }
 
       case LookupIterator::ACCESSOR: {
