diff --git a/build/install-build-deps.py b/build/install-build-deps.py
index b743710fd..86ace51d4 100755
--- a/build/install-build-deps.py
+++ b/build/install-build-deps.py
@@ -588,7 +588,6 @@ def backwards_compatible_list(options):
       "msttcorefonts",
       "python-dev",
       "python-setuptools",
-      "snapcraft",
       "ttf-dejavu-core",
       "ttf-indic-fonts",
       "ttf-kochi-gothic",
