diff --git a/BUILD.gn b/BUILD.gn
index d8e73f15932..d3324c04653 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -857,11 +857,7 @@ config("internal_config") {
     ":cppgc_header_features",
   ]
 
-  if (is_component_build) {
-    defines += [ "BUILDING_V8_SHARED_PRIVATE" ]
-  } else if (v8_expose_public_symbols) {
-    defines += [ "BUILDING_V8_SHARED" ]
-  }
+  defines += [ "BUILDING_V8_SHARED" ]
 
   if (v8_current_cpu == "riscv64" || v8_current_cpu == "riscv32") {
     if (!is_clang) {
@@ -6621,7 +6617,7 @@ v8_source_set("torque_ls_base") {
   ]
 }
 
-v8_component("v8_libbase") {
+v8_source_set("v8_libbase") {
   sources = [
     "src/base/abort-mode.cc",
     "src/base/abort-mode.h",
@@ -6792,11 +6788,7 @@ v8_component("v8_libbase") {
 
   data_deps = []
 
-  defines = []
-
-  if (is_component_build) {
-    defines = [ "BUILDING_V8_BASE_SHARED" ]
-  }
+  defines = [ "BUILDING_V8_BASE_SHARED" ]
 
   if (is_posix || is_fuchsia) {
     sources += [
@@ -6967,7 +6959,7 @@ if (v8_use_libm_trig_functions) {
   }
 }
 
-v8_component("v8_libplatform") {
+v8_source_set("v8_libplatform") {
   sources = [
     "include/libplatform/libplatform-export.h",
     "include/libplatform/libplatform.h",
@@ -7002,9 +6994,7 @@ v8_component("v8_libplatform") {
   v8_add_configs += [ "//build/config/compiler:thinlto_optimize_max" ]
   v8_remove_configs += [ "//build/config/compiler:thinlto_optimize_default" ]
 
-  if (is_component_build) {
-    defines = [ "BUILDING_V8_PLATFORM_SHARED" ]
-  }
+  defines = [ "BUILDING_V8_PLATFORM_SHARED" ]
 
   public_configs = [ ":libplatform_config" ]
 
@@ -7452,6 +7442,23 @@ if (v8_enable_webassembly) {
   }
 }
 
+v8_loadable_module("libv8android") {
+  defines = [ "BUILDING_V8_SHARED" ]
+
+  sources = [
+    "src/utils/v8dll-main.cc",
+  ]
+
+  deps = [
+    ":v8",
+    ":v8_libbase",
+    ":v8_libplatform",
+    ":v8_snapshot",
+  ]
+
+  configs = [ ":internal_config" ]
+}
+
 ###############################################################################
 # Executables
 #
diff --git a/gni/v8.gni b/gni/v8.gni
index 995e14cd82d..a70fb5fcc00 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -524,3 +524,12 @@ template("v8_static_library") {
     configs += invoker.configs
   }
 }
+
+template("v8_loadable_module") {
+  loadable_module(target_name) {
+    forward_variables_from(invoker, "*", [ "configs" ])
+    configs -= v8_remove_configs
+    configs += v8_add_configs
+    configs += invoker.configs
+  }
+}
