diff --git a/src/objects/js-objects.cc b/src/objects/js-objects.cc
index d1bc4cf4549..8aa5dc4903c 100644
--- a/src/objects/js-objects.cc
+++ b/src/objects/js-objects.cc
@@ -4453,26 +4453,26 @@ Maybe<bool> JSObject::PreventExtensionsWithTransition(
                    NewTypeError(MessageTemplate::kCannotFreeze));
   }
 
-  if (object->map()->has_named_interceptor() ||
-      object->map()->has_indexed_interceptor() ||
-      (object->HasTypedArrayOrRabGsabTypedArrayElements() &&
-       Cast<JSTypedArray>(*object)->IsVariableLength())) {
-    MessageTemplate message = MessageTemplate::kNone;
-    switch (attrs) {
-      case NONE:
-        message = MessageTemplate::kCannotPreventExt;
-        break;
-
-      case SEALED:
-        message = MessageTemplate::kCannotSeal;
-        break;
-
-      case FROZEN:
-        message = MessageTemplate::kCannotFreeze;
-        break;
-    }
-    RETURN_FAILURE(isolate, should_throw, NewTypeError(message));
-  }
+  // if (object->map()->has_named_interceptor() ||
+  //     object->map()->has_indexed_interceptor() ||
+  //     (object->HasTypedArrayOrRabGsabTypedArrayElements() &&
+  //      Cast<JSTypedArray>(*object)->IsVariableLength())) {
+  //   MessageTemplate message = MessageTemplate::kNone;
+  //   switch (attrs) {
+  //     case NONE:
+  //       message = MessageTemplate::kCannotPreventExt;
+  //       break;
+
+  //     case SEALED:
+  //       message = MessageTemplate::kCannotSeal;
+  //       break;
+
+  //     case FROZEN:
+  //       message = MessageTemplate::kCannotFreeze;
+  //       break;
+  //   }
+  //   RETURN_FAILURE(isolate, should_throw, NewTypeError(message));
+  // }
 
   DirectHandle<Symbol> transition_marker;
   if (attrs == NONE) {
